### PR TITLE
docs(DOMAPI) improve example on Location page showing that it's interactive

### LINKED ISSUE #21650: Live sample is not themed correctly
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Location

### What specific section or headline is this issue about?

Location anatomy

### What information was incorrect, unhelpful, or incomplete?

live sample is impossible/hard to read
image

### What did you expect to see?

a themed live sample that is easy to read

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/location`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Location
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/location/index.md
* Last commit: https://github.com/mdn/content/commit/bbecba9e7d688493adbdc37fc70e02d87adfe371
* Document last modified: 2022-10-10T08:33:38.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
This PR contains the following changes:

* Add description in example that it's interactive
* Add gray font for segments by default.
* Fix horizontal scrollbar
* Run Prettier & <strike>fix closing `>` tags</strike>

__Related issues:__

Fixes https://github.com/mdn/content/issues/21650

### CODE REVIEW

File: files/en-us/web/api/location/index.md [hunk 1]
```diff
@@ -99,12 +106,12 @@ body {
 ```
 
 ```js hidden
-document.body.addEventListener('click', (evt) => {
-    evt.preventDefault();
+document.body.addEventListener("click", (evt) => {
+  evt.preventDefault();
```
Comment (auchynnikau) [inline files/en-us/web/api/location/index.md:110]: Just a minor naming improvement :)

```suggestion
document.body.addEventListener("click", (event) => {
  event.preventDefault();
```

File: files/en-us/web/api/location/index.md [hunk 2]
```diff
@@ -99,12 +106,12 @@ body {
...
 
-    window.location.hash = evt.target.hasAttribute('id')
-        ? `#${evt.target.getAttribute('id')}`
-        : '';
+  window.location.hash = evt.target.hasAttribute("id")
+    ? `#${evt.target.getAttribute("id")}`
```
Comment (auchynnikau) [inline files/en-us/web/api/location/index.md:113]:
```suggestion
  window.location.hash = event.target.hasAttribute("id")
    ? `#${event.target.getAttribute("id")}`
```

File: files/en-us/web/api/location/index.md [hunk 3]
```diff
@@ -18,21 +18,19 @@ The **`Location`** interface represents the location (URL) of the object it is l
...
-  ><span id="search" title="search">?q=baz</span
-  ><span id="hash" title="hash">#bang</span></span
->
+<p>Hover over the URL segments below to highlight their meaning.</p>
+<span id="href" title="href">
+  <span id="origin" title="origin">
+    <span id="protocol" title="protocol">https:</span>//
+    <span id="host" title="host">
+      <span id="hostname" title="hostname">example.org</span>:
+      <span id="port" title="port">8080</span>
```
Comment (Josh-Cena) [inline files/en-us/web/api/location/index.md:27]: Just want to point out that HTML is whitespace sensitive so this introduces a space after `:`. I don't know if that's acceptable.

File: files/en-us/web/api/location/index.md [hunk 4]
```diff
@@ -49,6 +47,14 @@ body {
   font-size: 200%;
   line-height: 1em;
   white-space: nowrap;
+  overflow-x: hidden;
```
Comment (wbamberg) [inline files/en-us/web/api/location/index.md:50]: Rather than cut off the overflow, perhaps we could use a slightly smaller font, like `font-size: 180%`:

Screen Shot 2022-10-20 at 4 37 00 PM

(if it does overflow it's better to have a scrollbar than just to cut it off, but we should be able to avoid one at least with a full-width column.)

Comment (bsmth) [reply files/en-us/web/api/location/index.md:50]: I can certainly give it a shot

File: files/en-us/web/api/location/index.md (hunk 3 above)
Comment (bsmth) [reply files/en-us/web/api/location/index.md:27]: Very good, I think it makes sense not to add whitespace to the URL, I might revert the Prettier formatting in this case

Comment (bsmth) [reply files/en-us/web/api/location/index.md]: done in e57b45a0899a49a1e6d67c35692f9022e9b56a39

File: files/en-us/web/api/location/index.md (hunk 4 above)
Comment (bsmth) [reply files/en-us/web/api/location/index.md]: done in e57b45a0899a49a1e6d67c35692f9022e9b56a39

**Review by wbamberg — APPROVED** [review]
👍 nice, thank you @bsmth ! Leaving this open since there are other outstanding reviews.

File: files/en-us/web/api/location/index.md [hunk 5]
```diff
@@ -46,17 +47,25 @@ body {
   text-align: center;
   vertical-align: middle;
   font-family: Georgia;
-  font-size: 200%;
+  font-size: 175%;
```
Comment (Josh-Cena) [inline files/en-us/web/api/location/index.md:50]: FWIW all text here (including the text on the bottom) is a bit small. What's the motivation for this change? In fact I would prefer to enlarge the size of `[title]` a bit.

Comment (bsmth) [reply files/en-us/web/api/location/index.md:50]: >all text here (including the text on the bottom) is a bit small

That's true, especially the text on the bottom. @wbamberg had a suggestion to use smaller text instead of adding `overflow-x: hidden;` style to get rid of the horizontal scroll.

>I would prefer to enlarge the size of [title] a bit.

I think we should try this. I'll take another shot at it it to see if there's a better way to get rid of the horizontal scrollbar, keeping the text legible.

Comment (bsmth) [reply files/en-us/web/api/location/index.md:50]: See changes in 7ff0df5330fe98b4991c3eacd66f4393396d7d0c that I think works okay

Comment (wbamberg) [reply files/en-us/web/api/location/index.md:50]: You're right, it is too small. I wanted to avoid the scrollbar that is visible in https://github.com/mdn/content/issues/21650, but it turns out that scrollbar isn't visible at full width anyway, so it's not really a problem.

But I did just notice that the "Hover over the URL segments below to highlight their meaning." bit fades out when you hover, is that intentional?

**Review by Josh-Cena — APPROVED** [review]
This looks great to me

File: files/en-us/web/api/location/index.md (hunk 5 above)
Comment (bsmth) [reply files/en-us/web/api/location/index.md:50]: >"Hover over the URL segments below to highlight their meaning." bit fades out when you hover, is that intentional?

It is not 😄 this should be fixed

Comment (bsmth) [reply files/en-us/web/api/location/index.md:51]: I've just prefaced the example with the sentence instead of fighting the hover selector: https://github.com/mdn/content/pull/21678/commits/7eebebe11c906337aceecf825173505fd1d3e3c7

**Comment by bsmth** [comment]
Thanks both, going to merge 👍🏻
